src/serac/physics/utilities/functional/tensor.hpp

```diff
@@ -100,12 +100,58 @@ struct tensor<T> {
     return value;
   }
 
-  SERAC_HOST_DEVICE tensor() : value{} {}
-  SERAC_HOST_DEVICE tensor(T v) : value(v) {}
-  SERAC_HOST_DEVICE operator T() { return value; }
+  SERAC_HOST_DEVICE constexpr tensor() : value{} {}
+  SERAC_HOST_DEVICE constexpr tensor(T v) : value(v) {}
+  SERAC_HOST_DEVICE constexpr operator T() { return value; }
   T                 value;
 };
 
+template <typename T>
+struct tensor<T, 1> {
+  using type                     = T;
+  static constexpr int ndim      = 1;
+  static constexpr int first_dim = 1;
+
+  template <typename S>
+  SERAC_HOST_DEVICE constexpr auto& operator()(S, S) { return value; }
+
+  template <typename S>
+  SERAC_HOST_DEVICE constexpr auto operator()(S, S) const { return value; }
+
+  SERAC_HOST_DEVICE constexpr auto& operator[](int) { return value; };
+  SERAC_HOST_DEVICE constexpr auto  operator[](int) const { return value; };
+
+  SERAC_HOST_DEVICE constexpr operator T(){ return value; }
+  SERAC_HOST_DEVICE constexpr tensor() : value() {}
+  SERAC_HOST_DEVICE constexpr tensor(T v) : value(v) {}
+  T value;
+};
+
+template <typename T>
+struct tensor<T, 1, 1> {
+  using type                     = T;
+  static constexpr int ndim      = 2;
+  static constexpr int first_dim = 1;
+
+  template <typename S>
+  SERAC_HOST_DEVICE constexpr auto& operator()(S, S) { return value; }
+
+  template <typename S>
+  SERAC_HOST_DEVICE constexpr auto operator()(S, S) const { return value; }
+
+  SERAC_HOST_DEVICE constexpr auto& operator[](int) { return value; };
+  SERAC_HOST_DEVICE constexpr auto  operator[](int) const { return value; };
+
+  operator tensor<T, 1>(){ return value; }
+  tensor() : value() {}
+  tensor(T v) : value(v) {}
+  tensor(tensor< T, 1 > v) : value(v) {}
+
+  tensor< T, 1 > value;
+};
```

Comment:
these specializations help handle some of the degenerate cases that occur in the element_gradient_kernel when the test and/or trial spaces are scalar-valued